service/controller/v15/resource/deployment/master_deployment.go

```diff
@@ -327,7 +326,32 @@ func newMasterDeployments(customObject v1alpha1.KVMConfig) ([]*extensionsv1.Depl
 								ImagePullPolicy: apiv1.PullAlways,
 								Args: []string{
 									"daemon",
-									"--server.listen.address=http://127.0.0.1:60080",
+									"--server.listen.address=" + key.ShutdownDeferrerListenAddress(customObject),
+								},
+								Env: []apiv1.EnvVar{
+									{
+										Name: key.EnvKeyMyPodName,
+										ValueFrom: &apiv1.EnvVarSource{
+											FieldRef: &apiv1.ObjectFieldSelector{
+												FieldPath: "metadata.name",
+											},
+										},
+									},
+									{
+										Name: key.EnvKeyMyPodNamespace,
+										ValueFrom: &apiv1.EnvVarSource{
+											FieldRef: &apiv1.ObjectFieldSelector{
+												FieldPath: "metadata.namespace",
+											},
+										},
+									},
+								},
+								Lifecycle: &apiv1.Lifecycle{
+									PreStop: &apiv1.Handler{
+										Exec: &apiv1.ExecAction{
+											Command: []string{"/pre-shutdown-hook", fmt.Sprintf("%s/v1/defer/", key.ShutdownDeferrerListenAddress(customObject))},
```

Comment:
Now I see the whole picture. In a hindsight I think this whole `shutdown-deferer` repo could be put as another kvm-operator endpoint. We wouldn't have to maintain separate docker image.

Reply:
Original idea was to have reusable container that can be used for this purpose, but now that you say it, it is a bit dependent on the use case here so could make sense.